src/main/java/krona/command/AddCommand.java

```diff
@@ -0,0 +1,44 @@
+package krona.command;
+
+import krona.exception.KronaException;
+import krona.storage.Storage;
+import krona.task.Task;
+import krona.task.TaskList;
+import krona.ui.Ui;
+
+/**
+ * Represents a command to add a task to the task list.
+ */
+public class AddCommand extends Command {
+    private Task task;
+
+    /**
+     * Constructs an AddCommand with the specified task.
+     *
+     * @param task The task to be added to the task list.
+     */
+    public AddCommand(Task task) {
+        this.task = task;
+    }
+
+    /**
+     * Executes the command by adding the task to the task list, showing relevant messages,
+     * and saving the updated task list to storage.
+     *
+     * @param tasks The task list that the command operates on.
+     * @param ui The UI component that handles interactions with the user.
+     * @param storage The storage component that handles saving the updated task list.
+     * @throws KronaException If an error occurs while saving the task list.
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) throws KronaException {
+        tasks.addTask(task);
+        StringBuilder output = new StringBuilder();
+        output.append("Got it. I've added this task:\n");
```

Comment:
Would this belong in the purview of the UI as it is feedback to the user? Could consider.